kratos/tests/cpp_tests/utilities/test_parallel_utilities.cpp

```diff
@@ -0,0 +1,198 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Riccardo Rossi
+
+#include <utility>
+#include <iostream>
+#include "testing/testing.h"
+#include "utilities/parallel_utilities.h"
+#include "utilities/openmp_utils.h"
+
+namespace Kratos {
+namespace Testing {
+
+// Basic Type
+KRATOS_TEST_CASE_IN_SUITE(BlockPartitioner, KratosCoreFastSuite)
+{
+    int nsize = 1e3;
+    std::vector<double> data_vector(nsize);
+    for(auto& it : data_vector)
+        it = 5.0;
+
+    //here we raise every entry of a vector to the power 0.1
+    //c++11 version
+    BlockPartition<std::vector<double>>(data_vector).for_each(
+                                         [](double& item)
+    {
+        item = std::pow(item, 0.1);
+    });
+
+    //c++17 version - no explicit template parameter in constructor and auto in lambda
+    // BlockPartition(data_vector).for_each(
```

Comment:
To make it work before c++17, you could provide a helper function that would deduce BlockPartition type:
```cpp
template <class Vector>
BlockPartition<Vector> block_partition(Vector &&v) {
  return BlockPartition<Vector>(v);
}
```

Reply:
or even hide all this inside free functions (I like this):
```cpp
template <class Vector, class Func>
void block_for_each(Vector &&v, Func &&func) {
    BlockPartition<Vector>(std::forward<Vector>(v)).for_each(std::forward<Func>(func));
}
```

Reply:
mmm. the point is doing this hides the division between partitioning and for, which was one of the cool features.

at this point i could define the function as static and get along nicely.

@KratosMultiphysics/technical-committee  point of view on this?